scanner/src/main/kotlin/ScanController.kt

```diff
@@ -33,6 +33,7 @@ import org.ossreviewtoolkit.model.UnknownProvenance
 import org.ossreviewtoolkit.model.config.ScannerConfiguration
 import org.ossreviewtoolkit.scanner.provenance.NestedProvenance
 import org.ossreviewtoolkit.scanner.provenance.NestedProvenanceScanResult
+import org.ossreviewtoolkit.utils.common.PATH_STRING_COMPARATOR
```

Comment:
Why is it value-able to have this choice be deterministic?

(To my understanding, the only reason to choose a reference package is because the API to scan packages still accepts a `Package` as input, while it actually wants to have only a provenance. So, it does not matter what package is passed, because only it's provenance is considered).

Reply:
In the snippet choice feature that I am developing, the user writes a snippet choice in his/her `.ort.yml`.

The syntax is roughly:
```
package_snippet_choices:
  - id: "<package id>"
    snippet_choices:
    - license: "<snippet licenses>"
      reasoning: ""
      match:
        path: "<source file path>"
        snippet: "<snippet purl>"
```
Because the snippets are coming from a specific package scan.

The scanner correlates the snippet choice with the current scanner results, to adapt the license findings in consequence.
I guess same issue would arise for snippet curation, if we would like to fix the snippet findings.

But I agree, the day the scanners accept a provenance instead of a package, then this "feature" won't be needed anymore (and I need would adapt the snippet choice in consequence).

Anyway I think we will discuss it in details when I will present the snippet choice to ORT developers.

Reply:
If you want we can discuss if the migration to the provenance based interface for FossID makes sense.